west.yml: Update CMSIS-Core components to 5.8.0 release

stephanosio opened the pull request:
```
Update the CMSIS-Core components to the CMSIS 5.8.0 release:

* CMSIS-Core(M) 5.5.0
* CMSIS-Core(A) 1.2.1

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>
```

CMSIS PR: https://github.com/zephyrproject-rtos/cmsis/pull/11
~~STM32 PR: https://github.com/zephyrproject-rtos/hal_stm32/pull/113~~ (see https://github.com/zephyrproject-rtos/zephyr/pull/37697#issuecomment-904726253)

Closes #37693.

Depends on #37880.

stephanosio committed:
west.yml: Update CMSIS-Core components to 5.8.0 release

Update the CMSIS-Core components to the CMSIS 5.8.0 release:

* CMSIS-Core(M) 5.5.0
* CMSIS-Core(A) 1.2.1

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio commented:
STM32 HAL headers define their own `SCB_CFSR` fields and these are in conflict with the updated/corrected CMSIS definitions.
```
/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l1xx/soc/stm32l151xba.h:7797: error: "SCB_CFSR_MMARVALID_Pos" redefined [-Werror]
 7797 | #define SCB_CFSR_MMARVALID_Pos             (SCB_SHCSR_MEMFAULTACT_Pos + 7U)    /*!< SCB CFSR (MMFSR): MMARVALID Position */
      |
In file included from /var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l1xx/soc/stm32l151xba.h:133,
                 from /var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l1xx/soc/stm32l1xx.h:131,
                 from ../../../../../../soc/arm/st_stm32/stm32l1/soc.h:22,
                 from ../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:10,
                 from ../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:13,
                 from ../../../../../../include/arch/arm/aarch32/arch.h:187,
                 from ../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../include/kernel.h:17,
                 from /workdir/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /workdir/zephyr/arch/arm/core/offsets/offsets.c:9:
```

The duplicate definitions, as long as they are identical, are not problem; but, here the [new CMSIS release fixed incorrect definitions](https://github.com/ARM-software/CMSIS_5/commit/2e1a5d13866f7318aa5905b7fc8de69382cbd804) and these no longer match the definitions provided by the STM32 HAL (the latter still having incorrect definitions for these).

The incorrect STM32 HAL definitions should be updated accordingly to fix this issue.

cc @erwango @FRASTM

carlescufi commented:
@stephanosio could you please add a note to the release notes, under the "Architectures" section?

MaureenHelm reviewed (commented):
LGTM, thanks @stephanosio . Holding my official +1 until after the module PRs are merged and west.yml points to the new SHAs.

stephanosio committed:
doc: release-notes: Mention CMSIS version update

Mention that the CMSIS version has been updated to 5.8.0 in the Zephyr
2.7 release.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio commented:
**UPDATES**

* Removed the STM32 HAL update commit since the latest STM32Cube update (merged in https://github.com/zephyrproject-rtos/hal_stm32/pull/110) removed the `SCB_CFSR` definitions.
* Added release notes.

stephanosio commented:
Rebased after #37880 was merged.
